tests/unit_tests/core/tuple_accessor.cpp

```diff
@@ -0,0 +1,41 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#include <iostream>
+
+#include "detray/definitions/detail/accessor.hpp"
+
+// Thrust include(s).
+#include <thrust/tuple.h>
+
+// GoogleTest include(s).
+#include <gtest/gtest.h>
+
+TEST(tuple_accessor, tuple_accessor) {
+
+    using namespace detray;
+
+    // std::tuple test
+    auto s_tuple = detail::make_tuple<std::tuple>(1.0, 2, "std::tuple");
+
+    const int s_tuple_size = detail::tuple_size<decltype(s_tuple)>::value;
```

Comment:
Is that really ```int```?

Reply:
`auto` would indeed seem more appropriate. :smile:

Reply:
for std::tuple size, I think both `int` and `size_t` are ok
for thrust::tuple, `int` is right one

Reply:
I changed that :p